tensorly/tests/test_backend.py

```diff
@@ -1,47 +1,124 @@
-import numpy as np
-import tensorly as tl
+from concurrent.futures import ThreadPoolExecutor
 
-from scipy.sparse.linalg import svds
+import pytest
+import numpy as np
 from scipy.linalg import svd
 
+import tensorly as tl
 from ..backend import numpy_backend
 from .. import backend as T
 from ..base import fold, unfold
 from ..base import partial_fold, partial_unfold
 from ..base import tensor_to_vec, vec_to_tensor
 from ..base import partial_tensor_to_vec, partial_vec_to_tensor
+from ..testing import (assert_array_equal, assert_equal, assert_,
+                              assert_array_almost_equal, assert_raises)
 
 # Author: Jean Kossaifi
 
 
 def test_set_backend():
-    print('Testing set_backend for backend = {}'.format(tl._BACKEND))
+    torch = pytest.importorskip('torch')
+
+    toplevel_backend = tl.get_backend()
+
+    # Set in context manager
+    with tl.backend_context('numpy'):
+        assert tl.get_backend() == 'numpy'
+        assert isinstance(tl.tensor([1, 2, 3]), np.ndarray)
+        assert isinstance(T.tensor([1, 2, 3]), np.ndarray)
+        assert tl.float32 is T.float32 is np.float32
+
+        with tl.backend_context('pytorch'):
+            assert tl.get_backend() == 'pytorch'
+            assert torch.is_tensor(tl.tensor([1, 2, 3]))
+            assert torch.is_tensor(T.tensor([1, 2, 3]))
+            assert tl.float32 is T.float32 is torch.float32
+
+        # Sets back to numpy
+        assert tl.get_backend() == 'numpy'
+        assert isinstance(tl.tensor([1, 2, 3]), np.ndarray)
+        assert isinstance(T.tensor([1, 2, 3]), np.ndarray)
+        assert tl.float32 is T.float32 is np.float32
+
+    # Reset back to initial backend
+    assert tl.get_backend() == toplevel_backend
+
+    # Set not in context manager
+    try:
```

Comment:
Is the `try` needed here? Shouldn't the test just fail if an exception is raised anyway?

Reply:
Probably since we have `torch = pytest.importorskip('torch')` at the beginning.